File: app/src/main/java/com/mstc/mstcapp/ui/explore/project/ProjectViewModel.java
package com.mstc.mstcapp.ui.explore.project;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.mstc.mstcapp.Repository;
import com.mstc.mstcapp.model.explore.ProjectsModel;

import java.util.List;

public class ProjectViewModel extends AndroidViewModel {
    Repository repository;
    LiveData<List<ProjectsModel>> list;

    public ProjectViewModel(@NonNull Application application) {
        super(application);
        repository = new Repository(application);
        list = repository.getProjects();
    }

    public LiveData<List<ProjectsModel>> getList() {
        return list;
    }

    public void insertProjects(List<ProjectsModel> list) {
        repository.insertProjects(list);
    }

}

File: app/src/main/java/com/mstc/mstcapp/ui/explore/project/ProjectViewModel.kt
package com.mstc.mstcapp.ui.explore.project

import android.app.Application
import android.content.Context
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.mstc.mstcapp.data.ResourceRepository
import com.mstc.mstcapp.data.STCDatabase
import com.mstc.mstcapp.model.Result
import com.mstc.mstcapp.model.explore.Project
import kotlinx.coroutines.launch

private const val TAG = "ProjectViewModel"

class ProjectViewModel(application: Application) : AndroidViewModel(application) {
    val context: Context by lazy { application.applicationContext }
    private val repository =
        ResourceRepository(context, STCDatabase.getInstance(context))

    fun getProjects(): LiveData<Result<List<Project>>> {
        return repository.getProjects()
    }

    fun refreshProjects() {
        viewModelScope.launch { repository.refreshProjects() }
    }
}
